src/Data/Widget/Adapters/Adapter.elm

```diff
@@ -1,8 +1,14 @@
 module Data.Widget.Adapters.Adapter exposing (Adapter(..), decoder)
 
-import Data.Widget.Adapters.MetricAdapter as MetricAdapter exposing (Config, defaultConfig, adapterConfigDecoder)
-import Json.Decode as Decode exposing (Decoder, maybe)
+import Data.Widget.Adapters.CellPosition as CellPosition exposing (CellPosition, decoder)
+import Data.Widget.Adapters.MetricAdapter as MetricAdapter
+import Json.Decode as Decode exposing (Decoder, maybe, dict, string)
 import Json.Decode.Pipeline as Pipeline exposing (decode, required, optional)
+import Dict exposing (Dict)
+
+
+type alias Config =
+    Dict String CellPosition
```

Comment:
@msp I've had a stab at using a `Dict` for the config, but I'm having to define the value type as `CellPosition` upfront which I imagine will be troublesome later on.